finagle-thrift/src/main/scala/com/twitter/finagle/rich.scala

```diff
@@ -155,14 +157,25 @@ private[twitter] object ThriftUtil {
       } yield cons.newInstance(impl, protocolFactory)
 
     def tryScroogeFinagleService(iface: Class[_]): Option[BinaryService] =
-      for {
+      (for {
         baseName   <- findRootWithSuffix(iface.getName, "$FutureIface") orElse
           Some(iface.getName)
         serviceCls <- findClass[BinaryService](baseName + "$FinagleService") orElse
           findClass[BinaryService](baseName + "$FinagledService")
         baseClass  <- findClass1(baseName)
-        cons       <- findConstructor(serviceCls, baseClass, classOf[TProtocolFactory], classOf[StatsReceiver], Integer.TYPE)
-      } yield cons.newInstance(impl, protocolFactory, stats, Int.box(maxThriftBufferSize))
+      } yield {
```

Comment:
oh god.  thanks for doing this, this is crazy.  could you add a comment as to why we need this?